python-package/lightgbm/plotting.py

```diff
@@ -13,11 +13,11 @@
 def _check_not_tuple_of_2_elements(obj, obj_name='obj'):
     """Check object is not tuple or does not have 2 elements."""
     if not isinstance(obj, tuple) or len(obj) != 2:
-        raise TypeError('%s must be a tuple of 2 elements.' % obj_name)
+        raise TypeError(f"{obj_name} must be a tuple of 2 elements.")
 
 
 def _float2str(value, precision=None):
-    return ("{0:.{1}f}".format(value, precision)
+    return (f"{value:.{precision}f}"
```

Comment:
nice work! Did not know you could do string interpolation on the right-hand side of the `:` in an f-string.

I tested a few cases and confirmed that this still has the expected behavior.

```python
assert _float2str(0.0000001) == '1e-07'
assert _float2str(0.0000001, precision=1) == '0.0'
assert _float2str(0.0000001, precision=10) == '0.0000001000'
```

Reply:
Yes, even I learnt this today :)